src/Core/src/Hosting/MauiAppBuilder.cs

```diff
@@ -53,26 +53,21 @@ class MauiCoreInitializer : IMauiInitializeService
 			public void Initialize(IServiceProvider services)
 			{
 #if WINDOWS
-				var dispatcher = 
-					services.GetService<IDispatcher>() ??
-					MauiWinUIApplication.Current.Services.GetRequiredService<IDispatcher>();
+				if (UI.Xaml.Application.Current?.Resources is not UI.Xaml.ResourceDictionary resources)
+					return;
 
-				if (!dispatcher.IsDispatchRequired)
-					SetupResources();
+				if (resources.DispatcherQueue.HasThreadAccess)
+					SetupResources(resources);
 				else
-					dispatcher.Dispatch(SetupResources);
+					resources.DispatcherQueue.TryEnqueue(() => SetupResources(resources));
```

Comment:
Instead of using super complicated dispatcher logic, use the literally free dispatcher on the very dictionary we are adding to.

Reply:
Crashes in the very case it was designed to work in: https://github.com/microsoft/WindowsAppSDK/issues/2451